File: main/java/org/jskat/data/DesktopSavePathResolver.java
package org.jskat.data;

/**
 * Resolves the default save path for desktop applications
 */
public class DesktopSavePathResolver implements SavePathResolver {
	/**
	 * {@inheritDoc}
	 */
	@Override
	public String getDefaultSavePath() {
		return System.getProperty("user.home") //$NON-NLS-1$
				+ System.getProperty("file.separator") + ".jskat" //$NON-NLS-1$ //$NON-NLS-2$
				+ System.getProperty("file.separator"); //$NON-NLS-1$
	}
}

File: jskat-gui/src/main/kotlin/org/jskat/DesktopSavePathResolver.kt
package org.jskat

import org.jskat.data.SavePathResolver
import java.nio.file.Paths

class DesktopSavePathResolver : SavePathResolver {
    override fun getDefaultSavePath(): String {
        return Paths.get(System.getProperty("user.home"), ".jskat").toString()
    }

    override fun getCurrentWorkingDirectory(): String {
        return System.getProperty("user.dir")
    }
}
